src/Runtime/ServiceInstantiator.php

```diff
@@ -0,0 +1,204 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drush\Runtime;
+
+use Drush\Log\Logger;
+use League\Container\Container;
+use Drush\Drush;
+use Symfony\Component\Console\Application;
+use Composer\Autoload\ClassLoader;
+use Drush\Command\DrushCommandInfoAlterer;
+
+use Psr\Container\ContainerInterface;
+use League\Container\Container as DrushContainer;
+use Symfony\Component\Yaml\Yaml;
+use Robo\Robo;
+
+/**
+ * Use Dependency Injection Container to instantiate services.
+ */
+class ServiceInstantiator
+{
+    protected ContainerInterface $drushServicesContainer;
+    protected array $tags = [];
+
+    public function __construct(protected ContainerInterface $container, protected ContainerInterface $drushContainer)
+    {
+        $this->drushServicesContainer = new DrushContainer();
+    }
+
+    public function loadServiceFiles(array $serviceFiles)
+    {
+        foreach ($serviceFiles as $serviceFile) {
+            $serviceFileContents = '';
+            $serviceFileData = [];
+
+            if(file_exists($serviceFile)) {
+                $serviceFileContents = file_get_contents($serviceFile);
+            }
+            if (!empty($serviceFileContents)) {
+                $serviceFileData = Yaml::parse($serviceFileContents);
+            }
+
+            if (isset($serviceFileData['services'])) {
+                $this->instantiateServices($serviceFileData['services']);
+            }
+        }
+    }
+
+    /**
+     * Given a drush.services.yml file (parsed into an array),
+     * instantiate all of the services referenced therein, and
+     * cache them in our dynamic service container.
```

Comment:
"cache them" - the drush containers are rebuilt from scratch on every request, right? maybe we can say 'collect them' instead. if we are dumping containers to disk then we a similar cache rebuild issue that plagues the current module defined commands.

Reply:
Ah, yeah, there is no persistent cache here.